test/blackbox-tests/test-cases/directory-targets/copy-sandboxing.t

```diff
@@ -60,13 +60,13 @@ mode:
   permissions of output/
   755
   permissions of output/y
-  444
+  644
   permissions of output/x
-  444
+  644
   permissions of output/subdir
-  755
+  754
```

Comment:
I thought this PR only adds permissions. Here we are losing some?

Reply:
It's because we are now copying the permissions of directories correctly. Previously, we'd just do `Path.mkdir`, now we make sure we create the directory with the correct permissions.